provenance/repos.py

```diff
@@ -318,7 +318,13 @@ def __repr__(self):
             format(self.artifact.id, repr(self.__wrapped__))
 
     def __reduce__(self):
-        return (load_proxy , (self.artifact.id,))
+        return (load_proxy, (self.artifact.id,))
+
+    def __reduce_ex__(self, protocol_version):
+        return self.__reduce__()
+
+    def __copy__(self):
+        return ArtifactProxy(self.__wrapped__, self._self_artifact)
```

Comment:
For a `copy` I think we should be copying the `self.__wrapped__`. Otherwise if people modify the "copy" the original would be mutated as well. Adding a test around this behaviour would be helpful to make sure a copy is doing what we would expect it to.